content/en/docs/setup/install/multicluster/virtual-machine/index.md

```diff
@@ -0,0 +1,133 @@
+---
+title: Installing Istio for virtual machine connectivity
+description: Install an Isito mesh such that a virtual machine can connect to it and configure the virtual machine.
+weight: 3
+keywords:
+- kubernetes
+- virtual-machine
+- gateways
+- vms
+---
+
+Follow this guide to install an Istio control plane
+[multicluster deployment](/docs/ops/deployment/deployment-models/#multiple-clusters) and connect your first virtual machine.
+
+## Prerequisites
+
+- One or more Kubernetes clusters with versions: {{< supported_kubernetes_versions >}}.
+
+- Authority to [deploy the Istio control plane](/docs/setup/install/istioctl/).
+
+- Virtual machines must have IP connectivity to the ingress gateway in the connecting mesh, or alternatively, every pod in the mesh via L3 networking.
+
+- Services in the cluster must be accessible through the ingress gateway.
+
+## Installing the Istio control plane
+
+1. Generate intermediate CA certificates for each cluster's CA from your
+    organization's root CA. The shared root CA enables mutual TLS communication
+    across different clusters or virtual machines. For illustration purposes,
+    the following instructions use the certificates from the Istio samples
+    directory for the cluster.
+
+1. Run the following commands on each cluster in the mesh where a virtual
+     machine connection should be established. See
+     [Certificate Authority (CA) certificates](/docs/tasks/security/cert-management/plugin-ca-cert/)
+     for more details on configuring an external CA.
+
+{{< text bash >}}
+$ cd samples/certs
+$ make vm
+$ kubectl create namespace istio-system
+$ kubectl create secret generic cacerts -n istio-system \
+    --from-file=vm/ca-cert.pem \
+    --from-file=vm/ca-key.pem \
+    --from-file=vm/root-cert.pem \
+    --from-file=vm/cert-chain.pem
+$ cd ../..
+{{< /text >}}
+
+1. Generate a `cluster.env` configuration file that informs the virtual machine
+   deployment which network CIDR to capture and redirect to the Kubernetes
+   cluster:
+
+    {{< text bash >}}
+    $ ISTIO_SERVICE_CIDR=$(echo '{"apiVersion":"v1","kind":"Service","metadata":{"name":"tst"},"spec":{"clusterIP":"1.1.1.1","ports":[{"port":443}]}}' | kubectl apply -f - 2>&1 | sed 's/.*valid IPs is //')
+    $ echo -n ISTIO_SERVICE_CIDR=$ISTIO_SERVICE_CIDR > $HOME/cluster.env
+    {{< /text >}}
+
+1. If the Kubernetes cluster makes workload requests to the virtual machine,
+   the ports used in the request must be specified in `cluster.env`. An
+   example of exposing the `3306` and `8080` ports on the virtual machine:
+
+    {{< text bash >}}
+    $ echo "ISTIO_INBOUND_PORTS=3306,8080" >> $HOME/cluster.env
+    {{< /text >}}
+
+1. Install Istio with mesh expansion features:
+
+    {{< text bash >}}
```

Comment:
may be wise to convert this values-istio-meshexpansion.yaml into an IOP as is done in other parts of the documentation.

Reply:
+1

Also we should make it clear how to modify existing control plane install on the cluster to allow expansion, not just how to setup a new cluster with expansion.

For example I don't think the CA stuff actually is a requirement for mesh expansion. It really should just be "modify your control plane install to enable mesh expansion" and that's it.

Reply:
partially resolved by introducing an IOP.

I am unclear how the certificate creation works in virtual machine integration. @irisdingbj explained a CSR is created, then sent to istio, then Istio signs it, then that sends back a bundle, then that is installed in the VM. That is clearly not what is happening in this workflow, although that type of workflow may be happening under the hood.

@costinm had said that what I had explained wasn't the case in the Makefile review and in slack. For now, I am going to follow the guide outline that @rcaballeromx has constructed, and try to keep things simple - for his future integration of this work into his broader work of one doc for install every model.

The goal of this PR is to get virtual machine integration in a functional state so it can be evaluated (this feature has been touted as alpha for a long time, and broken since 1.4 after an archeology dig through the repository.

@irisdingbj when you begin your day and I end mine, would you mind taking a look at this comment and seeing if there is anything that can be done with this in the short term for 1.6?

Cheers,
-steve

Reply:
SGTM @smawson . resolving.